File: 365. Water and Jug Problem.cpp
class Solution {
public:
    int gcd(int a,int b)
    {
        return b==0? a: gcd(b,a%b);
    }
    bool canMeasureWater(int x, int y, int z) {
        if (z == 0) return true;
        if (z > x+y) return false;
        return z % gcd(x,y) == 0 ? true: false;
    }
};

File: 365. Water and Jug Problem.py
from fractions import gcd
class Solution(object):
    def canMeasureWater(self, x, y, z):
        """
        :type x: int
        :type y: int
        :type z: int
        :rtype: bool
        """
        if z == 0:
            return True
        if z > x+y:
            return False
        return True if z % gcd(x,y) == 0 else False
